### PR TITLE
Popup module update #68

### PR DESCRIPTION
**Pull request by anqaka**
#68 
https://dev.snowdog.pro/issues/35299

### CODE REVIEW

File: components/03-modules/popup/_popup.scss [hunk 1]
```diff
@@ -28,14 +33,30 @@ $popup__arrow-transform : rotate(45deg) !default;
         width: 100vw;
     }
 
-    &__arrow {
+    &__handler {
+        position: relative;
+        margin-top: $popup__content-top-margin;
+    }
+
+    &__close-btn {
```
Comment (kjugi) [inline components/03-modules/popup/_popup.scss]: Are we using shorted `btn` instead of `button` ?

File: components/03-modules/popup/popup.js [hunk 2]
```diff
@@ -1,33 +1,40 @@
...
       this.active = 'popup--active';
-
-      this.arrow = document.createElement('div');
-      this.arrow.classList.add('popup__arrow');
-      this.arrowSize = 21;
-
-      this.popup.appendChild(this.arrow);
-      this.arrowSelector = this.popup.querySelector('.popup__arrow');
+      this.body = document.querySelector('body');
+      this.closeButton = this.popup.querySelector('.popup__close-btn');
```
Comment (kjugi) [inline components/03-modules/popup/popup.js]: can we use here alph align or some other sorting? And maybe some formatting

File: components/03-modules/popup/popup.js [hunk 3]
```diff
@@ -41,69 +48,53 @@
...
-      this.isOutOfscreen(buttonCoords);
+      this.isOutOfscreen();
     }
 
-    isOutOfscreen(buttonCoords) {
+    isOutOfscreen() {
       const popupContentCoords = this.popup.getBoundingClientRect();
 
       if (window.matchMedia('(max-width: 480px)').matches) {
         const calcLeft = this.button.getBoundingClientRect().left * -1;
```
Comment (kjugi) [inline components/03-modules/popup/popup.js]: this -1px is still necessary? If it's referenced to border we have 0px

File: components/03-modules/popup/popup.js [hunk 4]
```diff
@@ -41,69 +48,53 @@
...
 
         this.popup.classList.add('popup--full-width');
         this.popup.style.setProperty('left', `${calcLeft}px`, '');
-
-        this.arrowSetPosition(buttonCoords.width, true);
       }
       else if (popupContentCoords.right >= this.body.clientWidth) {
         this.popup.classList.remove('popup--full-width');
```
Comment (kjugi) [inline components/03-modules/popup/popup.js]: we have `this.active` and there is modifier. Can we add this one too?

**Comment by kjugi** [comment]
I've noticed some weird behaviour when we have some i.e. text before button popup. It's not calculated property a top value. Please check locally if it happened too

File: components/03-modules/popup/_popup.scss (hunk 1 above)
Comment (anqaka) [reply components/03-modules/popup/_popup.scss]: IMO no difference if you knew what the element the class was described ;), changed for `button`

File: components/03-modules/popup/popup.js (hunk 3 above)
Comment (anqaka) [reply components/03-modules/popup/popup.js]: corrected

File: components/03-modules/popup/popup.js (hunk 4 above)
Comment (anqaka) [reply components/03-modules/popup/popup.js]: corrected

File: components/03-modules/popup/popup.js (hunk 2 above)
Comment (anqaka) [reply components/03-modules/popup/popup.js]: formatting added and sorting applied

**Comment by anqaka** [comment]
I corrected position function but I have to investigate some more option for positioning.

File: components/03-modules/popup/_popup.scss (hunk 1 above)
Comment (kjugi) [reply components/03-modules/popup/_popup.scss]: ok, thanks

**Pull request merged**